### DISCUSSION TITLE
Document Permissions

### DISCUSSION THREAD

**Discussion by claudinecombe** [pymupdf/PyMuPDF#2836] [Q&A] [answered]
Hello,

I am trying to set the document permissions at saving. I have found the different macro [here](https://pymupdf.readthedocs.io/en/latest/vars.html#document-permissions). 

I would like to enable printing with `PDF_PERM_PRINT` but prevent text selection (`PDF_PERM_COPY`) or document modification (`PDF_PERM_MODIFY`)

Thank you for your help 🙏

**Comment by JorjMcKie** [comment]
"Permisions" is an integer representing a bit field. You should start with a bit that always must be on, `PDF_PERM_ACCESSIBILITY` and then add more permissions via the "|" operator:
```python
perm = fitz.PDF_PERM_ACCESSIBILITY | fitz.PDF_PERM_PRINT | fitz.PDF_PERM_PRINT_HQ
doc.save(.., permissions=perm)
```
This leaves the other 5 possible permission bits off.

**Comment by claudinecombe** [comment]
Hi Jorj, thank you for your answer !

I did the above but I am still able to select text, I am wondering if this is happening since the new release of Adobe Acrobat Reader. 

Some time ago when I implemented it like the above I think it prevented from selecting text, but it does not anymore.

<img width="321" alt="image" src="https://github.com/pymupdf/PyMuPDF/assets/5883490/d525766f-6ffe-4c33-b10b-832e49a6a91f">

**Reply by JorjMcKie** [reply, marked as answer]
The situation is a bit ambiguous - I am sure you are aware.
First of all I forgot to mention that the save must include encrypting the file (with an owner pw and no user pw).
Then accessing the properties with e.g. Foxit Reader shows this:
![image](https://github.com/pymupdf/PyMuPDF/assets/8290722/44eec713-6a54-4bd8-a41b-84b4ac054792)
In my Adobe Acrobat and Foxit, I can still mark text on the page and press the <Ctrl>-copy key, but nothing in effect **_will_** be copied.
So a PDF viewer must be so kind and comply: my SumatraPDF for example still copies stuff!

**Reply by claudinecombe** [reply]
You are right, I can select text but no copy is made in Acrobat Reader, thank you ! 🙏

**Reply by SauravBhandari93** [reply]
Thank you for your help! Just a quick reminder:
↣**The User Password is used to open the PDF.**
↣**The Owner Password is used to set permissions, such as editing, printing, and copying.**
   To apply proper restrictions on the encrypted file, both passwords should be different.
